stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -1941,13 +1941,13 @@ void TaskGroupBase::waitAll(SwiftError* bodyError, AsyncTask *waitingTask,
 
   auto waitHead = waitQueue.load(std::memory_order_acquire);
   _swift_tsan_release(static_cast<Job *>(waitingTask));
+  if (!hasSuspended) {
+    waitingTask->flagAsSuspendedOnTaskGroup(asAbstract(this));
+    hasSuspended = true;
+  }
   while (true) {
-    if (!hasSuspended) {
-      hasSuspended = true;
-      waitingTask->flagAsSuspendedOnTaskGroup(asAbstract(this));
-    }
```

Comment:
Looks like we can also eliminate _this_ `hasSuspended` local boolean and just call `flagAsSuspendedOnTaskGroup` before the loop, as this is the only spot it's read.

Reply:
Yeah agreed; unconditionally setting it without the hasSuspended flat at all here seems okey to do.